nautobot/extras/models/groups.py

```diff
@@ -853,6 +863,54 @@ def get_group_members_url(self):
         """Return the group members URL."""
         return self.group.get_group_members_url()
 
+    def get_siblings(self, include_self=False):
+        """Return group memberships that share the same parent group."""
+        siblings = DynamicGroupMembership.objects.filter(parent_group=self.parent_group)
+        if include_self:
+            return siblings
+        return siblings.exclude(pk=self.pk)
+
+    # FIXME(jathan):
+    def _get_next_or_previous_by_weight(self, is_next):
+        """Get siblings and return the next/previous based on `is_next`."""
+        siblings = self.get_siblings(include_self=True)
+        if not is_next:
+            siblings = reversed(siblings)
+
+        siblings = list(siblings)
+        self_idx = siblings.index(self)
+
+        try:
+            return siblings[self_idx + 1]
+        except IndexError:
+            raise self.DoesNotExist(f"{self.__class__._meta.object_name} matching query does not exist.")
+
+    def get_next_by_weight(self):
+        """Get the next membership by weight."""
+        return self._get_next_or_previous_by_weight(is_next=True)
+
+    def get_previous_by_weight(self):
+        """Get the previous membership by weight."""
+        return self._get_next_or_previous_by_weight(is_next=False)
+
+    def generate_query(self):
+        return self.group.generate_query()
+
+    def calculate_next_query(self):
+        try:
+            prev_sib = self.get_previous_by_weight()
+        except self.DoesNotExist:
+            prev_sib = None
+
+        query = prev_sib.generate_query() if prev_sib is not None else models.Q()
+        next_set = self.generate_query()
+        return self.group.perform_membership_set_operation(self.operator, query, next_set)
+
+    def compute_member_count(self):
+        # query = self.group.generate_query()
+        next_query = self.calculate_next_query()
+        return self.group.model.objects.filter(next_query).count()
```

Comment:
Can we peel this out into it's own PR or stash it somewhere else? It seems this will be used to show the iterative application of filters which I don't see implemented in this PR.

The rest of this would be big value adds for the beta release of Dynamic Groups of Groups.

Reply:
I want to try to solve it. It's close, it's just not quite right and I'm currently prioritizing the other easier fixes.